File: KoreanTranslationCamera/KoreanTranslationCamera/Models/AppModels.swift
import Foundation

// MARK: - Supported Target Languages

enum SupportedLanguage: String, CaseIterable, Identifiable {
    case english = "en"
    case japanese = "ja"
    case chinese = "zh-Hans"

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .english: return "English"
        case .japanese: return "日本語"
        case .chinese: return "中文"
        }
    }

    var shortName: String {
        switch self {
        case .english: return "EN"
        case .japanese: return "JA"
        case .chinese: return "ZH"
        }
    }
}

// MARK: - Translation Result

struct TranslationResult: Equatable {
    let originalText: String
    let translatedText: String
    let targetLanguage: SupportedLanguage
    let timestamp: Date

    static func == (lhs: TranslationResult, rhs: TranslationResult) -> Bool {
        lhs.originalText == rhs.originalText &&
        lhs.targetLanguage == rhs.targetLanguage
    }
}

File: KoreanTranslationCameraAndroid/app/src/main/java/com/personal/koreantranslation/models/AppModels.kt
package com.personal.koreantranslation.models

import com.google.mlkit.nl.translate.TranslateLanguage
import java.util.Date

// ---------------------------------------------------------------------------
// Supported target languages
// ---------------------------------------------------------------------------

enum class SupportedLanguage(
    val displayName: String,
    val shortName: String,
    val mlKitCode: String
) {
    ENGLISH("English", "EN", TranslateLanguage.ENGLISH),
    JAPANESE("日本語",  "JA", TranslateLanguage.JAPANESE),
    CHINESE("中文",    "ZH", TranslateLanguage.CHINESE)
}

// ---------------------------------------------------------------------------
// Translation result
// ---------------------------------------------------------------------------

data class TranslationResult(
    val originalText: String,
    val translatedText: String,
    val targetLanguage: SupportedLanguage,
    val timestamp: Date = Date()
)

// ---------------------------------------------------------------------------
// Model download state (ML Kit downloads language packs on first use)
// ---------------------------------------------------------------------------

sealed class ModelDownloadState {
    object Idle        : ModelDownloadState()
    object Downloading : ModelDownloadState()
    object Ready       : ModelDownloadState()
    data class Error(val message: String) : ModelDownloadState()
}
